### PR TITLE
cleanup js api

### PR DESCRIPTION
**Pull request by hayes**
This cleans up the api exposed when used as a module.

I also added support for handling circular references and scoped modules

### CODE REVIEW

**Comment by jarofghosts** [comment]
once the readme is updated im totes :+1:

**Comment by jarofghosts** [comment]
might actually consider making the options argument optional

**Comment by hayes** [comment]
It is https://github.com/chrisdickinson/clone-packages/pull/16/files#diff-6d186b954a58d5bb740f73d84fe39073R7

**Comment by jarofghosts** [comment]
oh, derp, looked in the wrong spot

File: test/clone-recursive.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+      , 'data': content.toString('base64')
+      , 'length': content.length
+    }
+
+    data = JSON.stringify(data)
+
+    var options = {
+        headers: {
+            'content-type': 'application/json'
+          , 'content-length': new Buffer(data, 'utf8').length
```
Comment (hayes) [inline test/clone-recursive.js]: this should probably use `Buffer.byteLength`, oh well

Comment (jarofghosts) [inline test/clone-recursive.js]: just a test

**Pull request merged**